components/input/PasswordInput.js

```diff
@@ -1,24 +1,52 @@
 import React, { useState } from 'react';
+import { c } from 'ttag';
 import PropTypes from 'prop-types';
 import { Icon } from 'react-components';
 
-import Input from './Input';
+import { generateUID } from '../../helpers/component';
+import useInput from './useInput';
+import ErrorZone from '../text/ErrorZone';
 
-const PasswordInput = ({ disabled, ...rest }) => {
+const PasswordInput = (props) => {
     const [type, setType] = useState('password');
-    const toggle = () => setType(type === 'password' ? 'text' : 'password');
+
+    const toggle = () => {
+        setType(type === 'password' ? 'text' : 'password');
+    };
+
+    const { className, disabled, error, ...rest } = props;
+    const { handlers, statusClasses, status } = useInput(props);
+    const [uid] = useState(generateUID('passwordInput'));
 
     return (
         <>
-            <Input disabled={disabled} type={type} {...rest} />
-            <button disabled={disabled} type="button" onClick={toggle}>
-                <Icon name={type === 'password' ? 'read' : 'unread'} />
-            </button>
+            <span className="relative">
+                <input
+                    className={`pm-field w100 ${className} ${statusClasses}`}
+                    aria-invalid={error && status.isDirty}
+                    aria-describedby={uid}
+                    type={type}
+                    {...rest}
```

Comment:
Maybe `{...rest}` should be defined after `{...handlers}` to let the dev override it?

Reply:
No because useInput needs to take control of it, it'll be called if it's defined inside there